api/src/test/java/com/epam/pipeline/acl/docker/DockerRegistryApiServiceTest.java

```diff
@@ -0,0 +1,363 @@
+                .isEqualTo(dockerRegistry);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyUpdateDockerRegistryCredentialsForNonAdminUser() {
+        assertThrows(AccessDeniedException.class,
+            () -> dockerRegistryApiService.updateDockerRegistryCredentials(dockerRegistryVO));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldUpdateWholeDockerRegistryHierarchyCredentialsForAdmin() {
+        initDockerRegistryAclTree();
+        doReturn(dockerRegistryWithTools).when(mockDockerRegistryManager)
+                .updateDockerRegistryCredentials(dockerRegistryVO);
+
+        final DockerRegistry returnedDr = dockerRegistryApiService.updateDockerRegistryCredentials(dockerRegistryVO);
+
+        assertWholeDockerRegistryAclTree(returnedDr);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldListDockerRegistriesWithCerts() {
+        doReturn(dockerRegistryList).when(mockDockerRegistryManager).listAllDockerRegistriesWithCerts();
+
+        assertThat(dockerRegistryApiService.listDockerRegistriesWithCerts()).isEqualTo(dockerRegistryList);
+    }
+
+    @Test
+    @WithMockUser(SIMPLE_USER)
+    public void shouldListDockerRegistryHierarchyWithCerts() {
+        final DockerRegistryList dockerRegistryList = DockerCreatorUtils.getDockerRegistryList(dockerRegistry);
+        final DockerRegistry anotherRegistry = DockerCreatorUtils.getDockerRegistry(ID_3, ANOTHER_SIMPLE_USER);
+        dockerRegistryList.setRegistries(Arrays.asList(dockerRegistryWithTools, anotherRegistry));
+        initDockerRegistryAclTree();
+        doReturn(dockerRegistryList).when(mockDockerRegistryManager).listAllDockerRegistriesWithCerts();
+
+        final DockerRegistry returnedDr = (DockerRegistry) dockerRegistryApiService
+                .listDockerRegistriesWithCerts().getChildren().get(0);
+
+        assertPartialDockerRegistryAclTree(returnedDr);
+        assertThat(dockerRegistryApiService.listDockerRegistriesWithCerts().getChildren().size()).isEqualTo(1);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAllRegistriesContent() {
+        doReturn(dockerRegistryList).when(mockDockerRegistryManager).loadAllRegistriesContent();
+
+        assertThat(dockerRegistryApiService.loadAllRegistriesContent()).isEqualTo(dockerRegistryList);
+    }
+
+    @Test
+    @WithMockUser(SIMPLE_USER)
+    public void shouldLoadAllRegistriesHierarchyContent() {
+        final DockerRegistryList dockerRegistryList = DockerCreatorUtils.getDockerRegistryList(dockerRegistry);
+        final DockerRegistry anotherRegistry = DockerCreatorUtils.getDockerRegistry(ID_3, ANOTHER_SIMPLE_USER);
+        dockerRegistryList.setRegistries(Arrays.asList(dockerRegistryWithTools, anotherRegistry));
+        initDockerRegistryAclTree();
+        doReturn(dockerRegistryList).when(mockDockerRegistryManager).loadAllRegistriesContent();
+
+        final DockerRegistry returnedDr = (DockerRegistry) dockerRegistryApiService
```

Comment:
Could we store `dockerRegistryApiService.loadAllRegistriesContent().getChildren()` into the local field?